### PR TITLE
feat: The github platform supports AI capabilities

### PR DESCRIPTION
**Pull request by wwcchh0123**
### check list 
- [x] pr review comment support interactiving with reviewbot AI
- [x] issue comment support interactiving with reviewbot AI
- [x] The results of linter are explained in detail by reviewbot AI

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -67,38 +77,45 @@ require (
...
+	golang.org/x/exp v0.0.0-20230713183714-613f0c0eb8a1 // indirect
 	golang.org/x/net v0.28.0 // indirect
-	golang.org/x/oauth2 v0.19.0 // indirect
+	golang.org/x/oauth2 v0.20.0 // indirect
 	golang.org/x/sync v0.8.0 // indirect
 	golang.org/x/sys v0.25.0 // indirect
 	golang.org/x/term v0.23.0 // indirect
 	golang.org/x/text v0.17.0 // indirect
-	golang.org/x/time v0.3.0 // indirect
+	golang.org/x/time v0.5.0 // indirect
```
[the remark this answers is not in the document]
Comment (wwcchh0123) [reply go.mod:115]: @qiniu-x

File: server.go [hunk 2]
```diff
@@ -186,7 +203,8 @@ func (s *Server) handleGitHubEvent(ctx context.Context, event *github.PullReques
 	}
 
 	return s.withCancel(ctx, info, func(ctx context.Context) error {
-		installationID := event.GetInstallation().GetID()
+		// installationID := event.GetInstallation().GetID()
+		installationID := int64(56248106)
```
Comment (CarlJi) [inline server.go:207]: ?

File: main.go [hunk 3]
```diff
@@ -120,6 +128,12 @@ func gatherOptions() options {
 	fs.StringVar(&o.gitLabPersonalAccessToken, "gitlab.personal-access-token", "", "personal gitlab access token")
 	fs.StringVar(&o.gitLabHost, "gitlab.host", "", "gitlab server")
 
+	// llm related
+	fs.StringVar(&o.llmProvider, "llm.provider", "", "llm provider")
```
Comment (CarlJi) [inline main.go:132]: 简单规则的validate最好不下沉，因为也需要支持不配置的情况

Comment (wwcchh0123) [reply main.go:147]: updated

File: internal/linters/agent.go [hunk 4]
```diff
@@ -140,6 +145,11 @@ func (a *Agent) ApplyTypedMessageByIssueReferences(ctx context.Context, lintResu
 				}
 			}
 			if !processed {
+				resp, err := llm.QueryForReference(ctx, a.ModelClient, output.Message)
+				if err != nil {
+					log.Errorf("failed to query LLM server: %v", err)
```
Comment (CarlJi) [inline internal/linters/agent.go:150]: 应该需要continue

File: internal/linters/agent_test.go [hunk 5]
```diff
@@ -126,7 +127,7 @@ func TestApplyTypedMessageByIssueReferences(t *testing.T) {
 			expectedOutput: map[string][]LinterOutput{
 				"file4.go": {
 					{
-						Message: "regular lint message without issue reference",
+						Message: "regular lint message without issue reference" + fmt.Sprintf(ReferenceFooter, ""),
```
Comment (CarlJi) [inline internal/linters/agent_test.go:130]: 空的情况下，还有这个Footer有点奇怪

**Pull request merged**